File: .Net Components/XC-DMX512T/Samples/VB/NETComp VB Sample/Form1.vb
Public Class Form1
   
    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If IsNothing(XkDMXT_1.ConnectedDevices) Then
            deviceStatus.Text = "No XC-DMX512T Connected"
            deviceStatus.ForeColor = Color.Red
        Else
            deviceStatus.Text = "XC-DMX512T Connected"
            deviceStatus.ForeColor = Color.Green

            'Gets the Unit ID from the device
            lblUID.Text = "Unit ID: " & XkDMXT_1.ConnectedDevices(0).UnitID.ToString()

            'Gets the OEM Version ID from the device
            lblOEM.Text = "OEM Version ID: " & XkDMXT_1.ConnectedDevices(0).OEMVersion.ToString()

            'Gets the Product ID from the device
            lblProductID.Text = "Product ID: " & XkDMXT_1.ConnectedDevices(0).Pid.ToString()
        End If

    End Sub

    Private Sub Xk12SI_1_ButtonChange(ByVal e As XkDMXT.XKeyEventArgs) Handles XkDMXT_1.ButtonChange
        'This method handles the button change event for the device

        If e.PressState = True Then
            'button press
            Select Case e.CID
                Case 1001
                    lblButton01.Text = "Jack 1R-dn"
                    Exit Select
                Case 1002
                    lblButton02.Text = "Jack 1L-dn"
                    Exit Select
                Case 1003
                    lblButton03.Text = "Jack 2R-dn"
                    Exit Select
                Case 1004
                    lblButton04.Text = "Jack 2L-dn"
                    Exit Select
                Case 1005
                    lblButton05.Text = "Jack 3R-dn"
                    Exit Select
                Case 1006
                    lblButton06.Text = "Jack 3L-dn"
                    Exit Select
                Case 1007
                    lblButton07.Text = "Jack 4R-dn"
                    Exit Select
                Case 1008
                    lblButton08.Text = "Jack 4L-dn"
                    Exit Select
                Case 1009
                    lblButton09.Text = "Jack 5R-dn"
                    Exit Select
                Case 1010
                    lblButton10.Text = "Jack 5L-dn"
                    Exit Select
                Case 1011
                    lblButton11.Text = "Jack 6R-dn"
                    Exit Select
                Case 1012
                    lblButton12.Text = "Jack 6L-dn"
                    Exit Select
            End Select
        Else
            'button release
            Select Case e.CID
                Case 1001
                    lblButton01.Text = "Jack 1R-up"
                    Exit Select
                Case 1002
                    lblButton02.Text = "Jack 1L-up"
                    Exit Select
                Case 1003
                    lblButton03.Text = "Jack 2R-up"
                    Exit Select
                Case 1004
                    lblButton04.Text = "Jack 2L-up"
                    Exit Select
                Case 1005
                    lblButton05.Text = "Jack 3R-up"
                    Exit Select
                Case 1006
                    lblButton06.Text = "Jack 3L-up"
                    Exit Select
                Case 1007
                    lblButton07.Text = "Jack 4R-up"
                    Exit Select
                Case 1008
                    lblButton08.Text = "Jack 4L-up"
                    Exit Select
                Case 1009
                    lblButton09.Text = "Jack 5R-up"
                    Exit Select
                Case 1010
                    lblButton10.Text = "Jack 5L-up"
                    Exit Select
                Case 1011
                    lblButton11.Text = "Jack 6R-up"
                    Exit Select
                Case 1012
                    lblButton12.Text = "Jack 6L-up"
                    Exit Select
            End Select
        End If
        lblUID.Text = "Unit ID: " + XkDMXT_1.ConnectedDevices(0).UnitID.ToString()
       

    End Sub


    Private Sub btnUID_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnUID.Click
        Dim id As Integer = Convert.ToInt32(txtUID.Text)

        'Sets device Unit ID
        XkDMXT_1.SetDeviceUID(0, id)
        lblUID.Text = "Unit ID: " + XkDMXT_1.ConnectedDevices(0).UnitID.ToString()
    End Sub

    Private Sub btnOEM_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOEM.Click
        Dim id = Convert.ToInt32(txtOEM.Text)

        'Sets device OEM Version ID
        XkDMXT_1.SetOEMVersionID(0, id, True)
    End Sub

    

    Private Sub btnKeyboard_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnKeyboard.Click
        If (XkDMXT_1.ConnectedDevices(0).KeyboardInterface = True) Then
            txtKeyboard.Focus()
            'Sends the letters 'a' 'b' and 'c'
            XkDMXT_1.SendKeystrokes(0, False, False, False, False, False, False, False, False, 4, 5, 6, 0, 0, 0)
            'Releases all sent keys
            XkDMXT_1.SendKeystrokes(0, False, False, False, False, False, False, False, False, 0, 0, 0, 0, 0, 0)
        Else
            MessageBox.Show("No joystick endpoint available, change to pid 1195")
        End If
    End Sub

    Private Sub btnSetDongle_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSetDongle.Click
        'Enter in 4 bytes. REMEMBER these 4 bytes they will be required to check the dongle
        Dim byte1 As Byte = Convert.ToByte(txtByte1.Text)
        Dim byte2 As Byte = Convert.ToByte(txtByte2.Text)
        Dim byte3 As Byte = Convert.ToByte(txtByte3.Text)
        Dim byte4 As Byte = Convert.ToByte(txtByte4.Text)

        XkDMXT_1.SetSecurityKey(0, byte1, byte2, byte3, byte4)
    End Sub

    Private Sub btnCheckDongle_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCheckDongle.Click
        'Enter the 4 bytes used when XkDMXT_1.SetSecurityKey was called
        Dim byte1 As Byte = Convert.ToByte(txtByte1.Text)
        Dim byte2 As Byte = Convert.ToByte(txtByte2.Text)
        Dim byte3 As Byte = Convert.ToByte(txtByte3.Text)
        Dim byte4 As Byte = Convert.ToByte(txtByte4.Text)

        'Checks the security key by sending the selected bytes to the device; bytes were valid if it returns true
        Dim dongle As Boolean = XkDMXT_1.GetSecurityKey(0, byte1, byte2, byte3, byte4)

        If (dongle = True) Then
            grpDongle.BackColor = Color.Green
        ElseIf (dongle = False) Then
            grpDongle.BackColor = Color.Red
        End If
    End Sub

    Private Sub btnGenerate_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnGenerate.Click
        'Use this to get the state of the buttons even if there was no change in their states. This is often useful on start of application to get initial states of the buttons (switches).
        'Sending this command will trigger the _GenerateReportData event
        XkDMXT_1.GenerateReport(0)
    End Sub

    Private Sub btnFirmwareVersion_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnFirmwareVersion.Click
        lblFirmwareVersion.Text = XkDMXT_1.ConnectedDevices(0).FirmwareVersion.ToString()

    End Sub
    Private Sub XkDMXT_1_ButtonChange(ByVal e As XkDMXT.XKeyEventArgs) Handles XkDMXT_1.ButtonChange

    End Sub

    Private Sub XkDMXT_1_DevicePlug(ByVal e As XkDMXT.XKeyPlugEventArgs) Handles XkDMXT_1.DevicePlug
        deviceStatus.Text = "XC-DMX512T Plugged, Unit ID: " + e.UID.ToString() + ", OEM ID: " + e.OEMVersionID.ToString() + ", Product ID: " + e.PID.ToString()
        deviceStatus.ForeColor = Color.Green

        'Gets the Unit ID from the device
        lblUID.Text = "Unit ID: " + e.UID.ToString()

        'Gets the OEM Version ID from the device
        lblOEM.Text = "OEM ID: " + e.OEMVersionID.ToString()

        'Gets the Product ID from the device
        lblProductID.Text = "Product ID: " + e.PID.ToString()
    End Sub

    Private Sub XkDMXT_1_DeviceUnplug(ByVal e As XkDMXT.XKeyPlugEventArgs) Handles XkDMXT_1.DeviceUnplug
        deviceStatus.Text = "XC-DMX512T Unplugged, Unit ID: " + e.UID.ToString() + ", OEM ID: " + e.OEMVersionID.ToString() + ", Product ID: " + e.PID.ToString()
        deviceStatus.ForeColor = Color.Red

        lblUID.Text = "Unit ID: "
        lblOEM.Text = "OEM ID: "
        lblProductID.Text = "Product ID: "
    End Sub

    Private Sub XkDMXT_1_GenerateReportData(ByVal e As XkDMXT.XKeyEventArgs) Handles XkDMXT_1.GenerateReportData
        XkDMXT_1_ButtonChange(e)
    End Sub

    Private Sub btnSetSlots_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSetSlots.Click
        XkDMXT_1.SetNumberOfSlots(0, Convert.ToInt32(txtMaxSlots.Text))

    End Sub

    Private Sub btnSetSlotVals_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSetSlotVals.Click
        'example of setting all of the values of the slots
        Dim numberslots As Integer = XkDMXT_1.GetNumberOfSlots(0)
        For i As Integer = 0 To numberslots - 1
            'setting them all with a value of 128
            XkDMXT_1.SetSlotValue(0, i + 1, 128)
        Next
        'set the trackbar so it reflects the same value
        trackBar1.Value = XkDMXT_1.GetSlotValue(0, Convert.ToInt16(txtSlot.Text))
    End Sub

    Private Sub btnStart_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnStart.Click
        If btnStart.Text = "Start" Then
            btnStart.Text = "Stop"
            XkDMXT_1.TransmitContinuous(0, True)
        Else
            btnStart.Text = "Start"
            XkDMXT_1.TransmitContinuous(0, False)
        End If
    End Sub

    Private Sub trackBar1_Scroll(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles trackBar1.Scroll
        Dim slot As Integer = Convert.ToInt16(txtSlot.Text)
        Dim val As Byte = CByte(trackBar1.Value)
        lblSlotVal.Text = val.ToString()
        XkDMXT_1.SetSlotValue(0, slot, val)
    End Sub

    Private Sub btnTransmitOnce_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnTransmitOnce.Click
        'sending only 3 channels for this example
        XkDMXT_1.SetNumberOfSlots(0, 3)
        XkDMXT_1.SetSlotValue(0, 1, 0)
        'slot 1 = 0
        XkDMXT_1.SetSlotValue(0, 2, 255)
        'slot 2 = 255
        XkDMXT_1.SetSlotValue(0, 3, 0)
        'slot 3 = 0
        XkDMXT_1.Transmit(0)
    End Sub

    Private Sub chkGreen_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles chkGreen.CheckedChanged
        If chkGreen.Checked = False Then
            XkDMXT_1.SetGreenIndicator(0, 0)
        Else
            XkDMXT_1.SetGreenIndicator(0, 1)
        End If
    End Sub
End Class

File: .Net Components/XC-DMX512T/Samples/C#/NETComp Csharp Sample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace NETComp_Csharp_Sample
{
    
    public partial class Form1 : Form
    {
        
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            //When valid devices are connected they are added to the ConnectedDevices list
            //If this list is empty, then no valid devices are connected currently
           
            if (xkDMXT_1.ConnectedDevices == null)
            {
                deviceStatus.Text = "No XC-DMX512T Connected";
                deviceStatus.ForeColor = Color.Red;
            }
            else
            {
                deviceStatus.Text = "XC-DMX512T Connected";
                deviceStatus.ForeColor = Color.Green;

                //Gets the Unit ID from the device
                lblUID.Text = "Unit ID: " + xkDMXT_1.ConnectedDevices[0].UnitID;

                //Gets the OEM Version ID from the device
                lblOEM.Text = "OEM ID: " + xkDMXT_1.ConnectedDevices[0].OEMVersion;

                //Gets the Product ID from the device
                lblProductID.Text = "Product ID: " + xkDMXT_1.ConnectedDevices[0].Pid;
            }
        }

        private void HandleButtons(XkDMXT.XKeyEventArgs e)
        {
            //Handle state changes of the buttons
            if (e.PressState == true) //button press
            {
                switch (e.CID)
                {
                    case 1001:
                        lblButton01.Text = "Jack 1R-dn";
                        break;
                    case 1002:
                        lblButton02.Text = "Jack 1L-dn";
                        break;
                    case 1003:
                        lblButton03.Text = "Jack 2R-dn";
                        break;
                    case 1004:
                        lblButton04.Text = "Jack 2L-dn";
                        break;
                    case 1005:
                        lblButton05.Text = "Jack 3R-dn";
                        break;
                    case 1006:
                        lblButton06.Text = "Jack 3L-dn";
                        break;
                    case 1007:
                        lblButton07.Text = "Jack 4R-dn";
                        break;
                    case 1008:
                        lblButton08.Text = "Jack 4L-dn";
                        break;
                    case 1009:
                        lblButton09.Text = "Jack 5R-dn";
                        break;
                    case 1010:
                        lblButton10.Text = "Jack 5L-dn";
                        break;
                    case 1011:
                        lblButton11.Text = "Jack 6R-dn";
                        break;
                    case 1012:
                        lblButton12.Text = "Jack 6L-dn";
                        break;
                    
                }
            }
            else //button release
            {
                switch (e.CID)
                {
                    case 1001:
                        lblButton01.Text = "Jack 1R-up";
                        break;
                    case 1002:
                        lblButton02.Text = "Jack 1L-up";
                        break;
                    case 1003:
                        lblButton03.Text = "Jack 2R-up";
                        break;
                    case 1004:
                        lblButton04.Text = "Jack 2L-up";
                        break;
                    case 1005:
                        lblButton05.Text = "Jack 3R-up";
                        break;
                    case 1006:
                        lblButton06.Text = "Jack 3L-up";
                        break;
                    case 1007:
                        lblButton07.Text = "Jack 4R-up";
                        break;
                    case 1008:
                        lblButton08.Text = "Jack 4L-up";
                        break;
                    case 1009:
                        lblButton09.Text = "Jack 5R-up";
                        break;
                    case 1010:
                        lblButton10.Text = "Jack 5L-up";
                        break;
                    case 1011:
                        lblButton11.Text = "Jack 6R-up";
                        break;
                    case 1012:
                        lblButton12.Text = "Jack 6L-up";
                        break;
                }
            }
            lblUID.Text = "Unit ID: " + xkDMXT_1.ConnectedDevices[0].UnitID.ToString();
            
        }
        
        private void btnUID_Click(object sender, EventArgs e)
        {
            int id = Convert.ToInt32(txtUID.Text);

            //Sets device Unit ID
            xkDMXT_1.SetDeviceUID(0, id);
            lblUID.Text = "Unit ID: " + xkDMXT_1.ConnectedDevices[0].UnitID.ToString();
        }

        private void btnOEM_Click(object sender, EventArgs e)
        {
            int id = Convert.ToInt32(txtOEM.Text);

            //Sets device OEM Version ID
            xkDMXT_1.SetOEMVersionID(0, id, true);
            //after sending this command the device will disconnect and reconnect
        }

        private void btnKeyboard_Click(object sender, EventArgs e)
        {
            //Keyboard endpoint required therefore must be in Product ID 1195
            if (xkDMXT_1.ConnectedDevices[0].KeyboardInterface == true)
            {
                txtKeyboard.Focus();
                //Sends the letters 'a' 'b' and 'c'
                xkDMXT_1.SendKeystrokes(0, false, false, false, false, false, false, false, false, 4, 5, 6, 0, 0, 0);
                //Releases all sent keys
                xkDMXT_1.SendKeystrokes(0, false, false, false, false, false, false, false, false, 0, 0, 0, 0, 0, 0);
            }
            else
            {
                MessageBox.Show("No keyboard endpoint available, change to pid 1195");

            }
        }

        private void btnMouse_Click(object sender, EventArgs e)
        {
        }

        private void btnJoystick_Click(object sender, EventArgs e)
        {
           
        }

        private void btnCheckDongle_Click(object sender, EventArgs e)
        {
            //Enter the 4 bytes used when SetSecurityKey was called
            byte byte1 = Convert.ToByte(txtByte1.Text);
            byte byte2 = Convert.ToByte(txtByte2.Text);
            byte byte3 = Convert.ToByte(txtByte3.Text);
            byte byte4 = Convert.ToByte(txtByte4.Text);

            //Checks the security key by sending the selected bytes to the device; bytes were valid if it returns true
            bool dongle = xkDMXT_1.GetSecurityKey(0, byte1, byte2, byte3, byte4);

            if (dongle == true) //correct dongle was entered
            {
                grpDongle.BackColor = Color.Green;
            }
            if (dongle == false) //incorrect dongle was entered
            {
                grpDongle.BackColor = Color.Red;
            }
        }

        private void btnSetDongle_Click(object sender, EventArgs e)
        {
            //Enter in 4 bytes. REMEMBER these 4 bytes they will be required to check the dongle
            byte byte1 = Convert.ToByte(txtByte1.Text);
            byte byte2 = Convert.ToByte(txtByte2.Text);
            byte byte3 = Convert.ToByte(txtByte3.Text);
            byte byte4 = Convert.ToByte(txtByte4.Text);

            xkDMXT_1.SetSecurityKey(0, byte1, byte2, byte3, byte4);
        }

        private void btnGenerate_Click(object sender, EventArgs e)
        {
            //Use this to generate a report which gives the state of the buttons even if there was no change in their states, this is often useful on start of application to get initial states of the switches (buttons)
            xkDMXT_1.GenerateReport(0);
        }

        private void xkDMXT_1_ButtonChange(XkDMXT.XKeyEventArgs e)
        {
            HandleButtons(e);
        }

        private void xkDMXT_1_DevicePlug(XkDMXT.XKeyPlugEventArgs e)
        {
            deviceStatus.Text = "XC-DMX512T, Unit ID: " + e.UID.ToString() + ", OEM ID: " + e.OEMVersionID.ToString() + ", Product ID: " + e.PID.ToString();
            deviceStatus.ForeColor = Color.Green;
            
            lblUID.Text = "Unit ID: " + e.UID.ToString();
            lblOEM.Text = "OEM ID: " + e.OEMVersionID.ToString();
            lblProductID.Text = "Product ID: " + e.PID.ToString();
        }

        private void xkDMXT_1_DeviceUnplug(XkDMXT.XKeyPlugEventArgs e)
        {
            deviceStatus.Text = "XC-DMX512T Unplugged, Unit ID: " + e.UID.ToString() + ", OEM ID: " + e.OEMVersionID.ToString() + ", Product ID: " + e.PID.ToString();
            deviceStatus.ForeColor = Color.Red;

            lblUID.Text = "Unit ID: ";
            lblOEM.Text = "OEM ID: ";
            lblProductID.Text = "Product ID: ";
        }

        private void xkDMXT_1_GenerateReportData(XkDMXT.XKeyEventArgs e)
        {
            HandleButtons(e);
        }

        private void btnSetSlots_Click(object sender, EventArgs e)
        {
            xkDMXT_1.SetNumberOfSlots(0, Convert.ToInt16(txtMaxSlots.Text));
        }

        private void btnSetSlotVals_Click(object sender, EventArgs e)
        {
            //example setting all of the values of the slots
            int numberslots = xkDMXT_1.GetNumberOfSlots(0);
            for (int i = 0; i < numberslots; i++)
            {
                xkDMXT_1.SetSlotValue(0, i + 1, 128); //setting them all with a value of 128
            }
            //set the trackbar so it reflects the same value
            trackBar1.Value = xkDMXT_1.GetSlotValue(0, Convert.ToInt16(txtSlot.Text));

        }

        private void btnStart_Click(object sender, EventArgs e)
        {
            if (btnStart.Text == "Start")
            {
                btnStart.Text = "Stop";
                xkDMXT_1.TransmitContinuous(0, true);
            }
            else
            {
                btnStart.Text = "Start";
                xkDMXT_1.TransmitContinuous(0, false);
            }
        }

        private void trackBar1_ValueChanged(object sender, EventArgs e)
        {
            int slot = Convert.ToInt16(txtSlot.Text);
            byte val = (byte)trackBar1.Value;
            lblSlotVal.Text = val.ToString();
            xkDMXT_1.SetSlotValue(0, slot, val);
        }

        private void btnFirmwareVersion_Click(object sender, EventArgs e)
        {
            lblFirmwareVersion.Text = xkDMXT_1.ConnectedDevices[0].FirmwareVersion.ToString();
        }

        private void btnTransmitOnce_Click(object sender, EventArgs e)
        {
            //sending only 3 channels for this example
            xkDMXT_1.SetNumberOfSlots(0, 3);
            xkDMXT_1.SetSlotValue(0, 1, 0); //slot 1 = 0
            xkDMXT_1.SetSlotValue(0, 2, 255); //slot 2 = 255
            xkDMXT_1.SetSlotValue(0, 3, 0); //slot 3 = 0

            xkDMXT_1.Transmit(0);
        }

        
        


    }
}
